packages/cdktf/lib/terraform-stack.ts

```diff
@@ -129,18 +129,13 @@ export class TerraformStack extends Construct {
       }
     };
 
-    const visit = (node: IConstruct) => {
-      if (node instanceof TerraformElement) {
-        deepMerge(tf,  node.toTerraform());
-      }
+    const elements = terraformElements(this);
+    const fragments = elements.map(e => resolve(this, e.toTerraform()));
```

Comment:
do we have to call it for each element?

Reply:
Technically no; however, the act of merging clobbers information necessary to resolve an intrinsic, so this seems like the cleanest way of accomplishing that (also how AWS CDK does it). 

It might make sense to resolve the `rawOverrides`, but I wouldn't expect tokens to exist there.

I'm not entirely sure if the final `resolve` is necessary, but AWS CDK has it and it seems like a good safeguard.

Reply:
cool - thanks for elaborating 👍